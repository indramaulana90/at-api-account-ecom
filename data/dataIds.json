[
	{
		"relativePath": "__tests__/v1/example/post/example/data.js",
		"ids": {
			"idUsers": 999999999
		}
	},
	{
		"relativePath": "__tests__/v1/order/get/order_parcel_detail/data.js",
		"ids": {
			"idOrders": 839640065,
			"idOrderParcels": 839641572,
			"idOrderParcelProducts": 839644779,
			"idBrands": "888888f39970108f39970108",
			"idBuyers": 839966387,
			"idUrbans": 839966387,
			"idInvoiceGroups": "888888f39970108f39970108",
			"idCart": "888888f39970314f39970314",
			"idDevice": "839966387",
			"idProducts": "888888f39970108f39970108",
			"idPhoneNumber": 839981005
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_list/data.js",
		"ids": {
			"idWorkingDays": [
				841736409,
				841736511,
				841736535
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_detail/data.js",
		"ids": {
			"idWorkingDays": [
				841802581,
				841802625
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/workingDay/data.js",
		"ids": {
			"idWorkingDays": [
				841888623,
				841888635,
				841888672
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/update_workingday/data.js",
		"ids": {
			"idWorkingDays": [
				841909018,
				841909045,
				841909072
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/post/cancel_order_parcel/data.js",
		"ids": {
			"idOrders": 842063620,
			"idOrderParcels": 842063620,
			"idOrderParcelProducts": 842063620,
			"idBrands": "888888f42063620f42063620",
			"idBuyers": 842063620,
			"idUrbans": 842063620,
			"idInvoiceGroups": "888888f42063620f42063620",
			"idCart": "888888f42063620f42063620",
			"idDevice": 842063620,
			"idProducts": "888888f42063620f42063620",
			"idPhoneNumber": 842063620
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_list/data.js",
		"ids": {
			"idHolidays": [
				841986121,
				841986145,
				841986166,
				841986181,
				841986198
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_detail/data.js",
		"ids": {
			"idHolidays": [
				841994725,
				841994752
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/holidays/data.js",
		"ids": {
			"idHolidays": [
				841999606,
				841999633,
				841999652
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/post/holiday/data.js",
		"ids": {
			"idHolidays": 842147917
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/holiday_update/data.js",
		"ids": {
			"idHolidays": [
				842318652,
				842318674,
				842318692,
				842321145
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/patch/update_order_status/data.js",
		"ids": {
			"idOrders": 842412325,
			"idOrderParcels": 842412325,
			"idOrderParcelProducts": 842412325,
			"idBrands": "888888f42412325f42412325",
			"idBuyers": 842412325,
			"idUrbans": 842412325,
			"idInvoiceGroups": "888888f42412325f42412325",
			"idCart": "888888f42412325f42412325",
			"idDevice": 842412325,
			"idProducts": "888888f42412325f42412325",
			"idPhoneNumber": 842412325
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/order_sellers_is_read/data.js",
		"ids": {
			"idOrders": 856578262,
			"idOrderSellers": 856578315
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_failed/data.js",
		"ids": {
			"idOrders": 856642262,
			"idOrderSellers": 856642262,
			"idCart": "888888f56642550f56642550",
			"idDevice": 856642550,
			"idOrderBilling": 856642736,
			"idOrderSellerStatusHistoriesId": 856642736,
			"idOrderProduct": 856642736
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_cancel/data.js",
		"ids": {
			"idOrders": 856664266,
			"idOrderSellers": 856664266
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/post/assign_product/data.js",
		"ids": {
			"idWarehouses": 862452416,
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/patch/warehouse_assign_product/data.js",
		"ids": {
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	}
]