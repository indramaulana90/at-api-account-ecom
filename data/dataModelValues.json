{
    "banners":{
        "name": "ATAPI ${params.atName} Banner Name",
        "header": "ATAPI${params.atName} Header",
        "description": "ATAPI ${params.atName} Description",
        "termAndCondition": "<p>• Berlaku untuk user dari semua tier</p><p>• Pembelian min 1.000.000 untuk SKU (SGM 300gr)</p><p>• Tidak berlaku kelipatan</p>"
    },
    "brands": {
        "name": "ATAPI ${params.atName} Brand Name",
        "code": "ATAPI${params.atName}Brand"
    },
    "buyers": {
        "name": "ATAPI ${params.atName} Buyers Name",
        "address": "ATAPI ${params.atName} Address"
    },
    "catalogues": {
        "name": "ATAPI ${params.atName} Catalogue Name",
        "code": "ATAPI${params.atName}Cata",
        "detail": "ATAPI ${params.atName} Catalogue Detail",
        "description": "ATAPI ${params.atName} Catalogue Description",
        "information": "ATAPI ${params.atName} Catalogue Information"
    },
    "catalogueCategories": {
        "name": "ATAPI ${params.atName} Category Name",
        "code": "ATAPI${params.atName}CCat"
    },
    "catalogueUnits": {
        "name": "ATAPI ${params.atName} Catalogue Unit Name"
    },
    "channels": {
        "name": "ATAPI ${params.atName} Channel Name",
        "externalId": "ATAPI${params.atName}Chan",
        "description": "ATAPI ${params.atName} Description"
    },
    "clusters": {
        "name": "ATAPI ${params.atName} Cluster Name",
        "externalId": "ATAPI${params.atName}Clus",
        "description": "ATAPI ${params.atName} Description"
    },
    "groups": {
        "name": "ATAPI ${params.atName} Group Name",
        "externalId": "ATAPI${params.atName}Grup",
        "description": "ATAPI ${params.atName} Description"
    },
    "invoiceGroups": {
        "name": "ATAPI ${params.atName} Invoice Group name",
        "code": "ATAPI${params.atName}InGr",
        "externalId": "ATAPI${params.atName}InGr"
    },
    "mssCatalogues": {
        "name": "ATAPI ${params.atName} MssCatalogues Name",
        "externalId": "ATAPI${params.atName}MSS"
    },
    "portfolios": {
        "name": "ATAPI ${params.atName} Portfolio Name",
        "externalId": "ATAPI${params.atName}Ptfl"
    },
    "products": {
        "name": "ATAPI ${params.atName} Product Name",
        "externalId": "ATAPI${params.atName}Pdct"
    },
    "stores": {
        "address": "ATAPI ${params.atName} Stores Address",
        "name": "ATAPI ${params.atName} Stores Name",
        "noteAddress": "ATAPI ${params.atName} Stores Note Address",
        "externalId": "ATAPI${params.atName}Strs"
    },
    "salesRep": {
        "name": "ATAPI ${params.atName} Users Sales fullname",
        "externalId": "ATAPI${params.atName}UsSl"
    },
    "salesTeamStoreChannels": {
        "name": "ATAPI ${params.atName} ST Store Channels Name"
    },
    "types": {
        "name": "ATAPI ${params.atName} Type Name",
        "externalId": "ATAPI${params.atName}Type",
        "description": "ATAPI ${params.atName} Description"
    },
    "urbans": {
        "city": "ATAPI ${params.atName} City",
        "district": "ATAPI ${params.atName} District",
        "urban": "ATAPI ${params.atName} Urban"
    },
    "users": {
        "fullname": "ATAPI ${params.atName} users fullname",
        "email":"ATAPI${params.atName}@atapi.com",
        "userCode": "ATAPI${params.atName}Usrs"
    },
    "usersSales": {
        "fullname": "ATAPI ${params.atName} users sales fullname",
        "email":"ATAPI${params.atName}@sales.com",
        "userCode": "ATAPI${params.atName}UsSl"
    },
    "vouchers": {
        "name": "ATAPI {params.atName} Name",
        "externalId": "ATAPI${params.atName}Vouc",
        "code": "${params.atName}",
        "description": "ATAPI ${params.atName} Description",
        "header":"ATAPI ${params.atName} Header",
        "terms": "{ATAPI ${params.atName} Terms and Condition}",
        "instructions": "{ATAPI ${params.atName} Instructions}",
        "tag": "{ATAPI ${params.atName} Tag}"
    },
    "warehouses": {
        "name": "ATAPI ${params.atName} Warehouse name",
        "code": "ATAPI${params.atName}WH",
        "externalId": "ATAPI${params.atName}WH",
        "noteAddress": "ATAPI ${params.atName} Note Address",
        "address": "ATAPI ${params.atName} Address"
    },
    "warehouseValues": {
        "name": "ATAPI ${params.atName} Warehouse Value name"
    },
    "warehouseTemperatures": {
        "name": "ATAPI ${params.atName} Warehouse Temperature name"
    }
}