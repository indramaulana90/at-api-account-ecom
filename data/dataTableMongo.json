[
	{
		"idInvoiceGroups": {
			"isMongo": true
		}
	},
	{
		"idInvoiceGroupBrands": {
			"isMongo": true
		}
	},
	{
		"idBrands": {
			"isMongo": true
		}
	},
	{
		"idProducts": {
			"isMongo": true
		}
	},
	{
		"idCart": {
			"isMongo": true
		}
	}
]
