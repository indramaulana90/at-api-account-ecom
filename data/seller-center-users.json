[
  { "name": "TRS", "username": "admin.trs@sinbad.co.id", "password": "sinbad" },
  { "name": "Unipro", "username": "admin.unipro@sinbad.co.id", "password": "sinbad" },
  { "name": "Quality", "username": "quality.sinbad@gmail.com", "password": "quality3607" },
  { "name": "Preprod", "username": "admin.techquality@sinbad.co.id", "password": "sinbad" }
]
