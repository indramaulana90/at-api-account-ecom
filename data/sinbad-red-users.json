[
    { "name": "red1", "mobilePhone": "082297872220"},
    { "name": "Scimitar-SR", "mobilePhone": "081239999999", "email": "ScimitarSR@ATAPIqe.com" },
    { "name": "Kanzun-SR", "mobilePhone": "081238888888", "email": "KanzunSR@ATAPIqe.com" },
    { "name": "Voyager-SR", "mobilePhone": "081237777777", "email": "VoyagerSR@ATAPIqe.com" },
    { "name": "Dinar-SR", "mobilePhone": "081236666666", "email": "DinarSR@ATAPIqe.com" },
    { "name": "Valkyrie-SR", "mobilePhone": "08786666666", "email": "ValkyrieSR@ATAPIqe.com" },
    { "name": "Valkyrie-Stg", "mobilePhone": "08783333333", "email": "ValkyrieStg@ATAPIqe.com" },
    { "name": "Pyramid-SR", "mobilePhone": "081234444444", "email": "PyramidSR@ATAPIqe.com" },
    { "name": "DavyJones-SR", "mobilePhone": "081232222222", "email": "DavyJonesSR@ATAPIqe.com" },
    { "name": "Chimera-SR", "mobilePhone": "081231111111", "email": "ChimeraSR@ATAPIqe.com" },
    { "name": "No Name", "mobilePhone": "085717216917", "email": "chimera@mail.com" }
  ]
  