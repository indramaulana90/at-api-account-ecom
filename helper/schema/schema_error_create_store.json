{
    "type": "object",
    "properties": {
      "status": {
        "type": "integer"
      },
      "message": {
        "type": "string"
      },
      "errorMessage": {
        "type": "string"
      },
      "type": {
        "type": "string"
      },
      "code": {
        "type": "integer"
      }
    },
    "additionalProperties": false,
    "required": [
      "status",
      "message",
      "errorMessage",
      "type",
      "code"
    ]
  }