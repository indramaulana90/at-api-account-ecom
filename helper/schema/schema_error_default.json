{
  "type": "object",
  "properties": {
      "message": {
          "type": "string"
      },
      "errorMessage": {
          "type": ["string", "object"]
      },
      "type": {
          "type": "string"
      },
      "code": {
          "type": "integer"
      }
  },
  "additionalProperties": false,
  "required": [
      "message",
      "errorMessage",
      "type",
      "code"
  ]
}