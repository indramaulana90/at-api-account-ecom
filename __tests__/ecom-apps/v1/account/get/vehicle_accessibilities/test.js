/**
* Author      : Fauzan Novaldi Suteja (fauzan.novaldi@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This Main File for Testing on API get_vehicle_accessibilities
*/

import getDataFromPool from '@at-api-core/get-data-from-pool';
import * as er from '@at-api-core/expect-response';
import endpoint from './endpoint';
import * as data from './data';
import tc from './tc';
import schema from './schema.json';
import rs from '../../../../../../helper/responseStatusLibrary.json';

const { cookies } = getDataFromPool('ng','sinbad-app', 'Chimera-SR_081231111111');

describe(`${tc.type[1]} ${tc.describe}`, () => {
  test(`${tc.positive.getListVehAccess}`, async () => {
    resp = await endpoint(cookies);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);
    er.expectSnapshot(resp);
  });

  test(`${tc.positive.paramKeyword}`, async () => {
    const payload = data.params('motor');
    resp = await endpoint(cookies, payload);
    const objectResponse = resp.body.data[0];
    const name = (objectResponse.name).toLowerCase();
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);
    er.expectContain(resp, name, payload.keyword);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.keywordNotFound}`, async () => {
    const payload = data.params('bus');
    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);
    er.expectSnapshot(resp);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.xPlatformNull}`, async () => {
    resp = await endpoint(cookies).set('x-platform', null);
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformEmpty}`, async () => {
    resp = await endpoint(cookies).unset('x-platform', '');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformInvalid}`, async () => {
    resp = await endpoint(cookies).set('x-platform', 'error');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformNotSet}`, async () => {
    resp = await endpoint(cookies).unset('x-platform');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  data.loopingErrorCookies.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const cookiesNegative = element.value;

      resp = await endpoint(cookiesNegative);
      er.expectStatus(resp, rs.statusUnauthorized);
      er.expectUnauthorized(resp);
      er.expectSnapshot(resp);
    });
  });
});
