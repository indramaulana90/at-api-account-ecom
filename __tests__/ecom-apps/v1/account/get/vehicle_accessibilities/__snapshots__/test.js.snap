// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get all list of bank with x-platform empty 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get all list of bank with x-platform null 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities because cookies set to Empty 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities because cookies set to Invalid 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities because cookies set to Null 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities because set invalid on x-platform header 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities because x-platform header is not set 1`] = `Object {}`;

exports[`@negative get vehicle_accessibilities -- As a Store, I should not be able to get list of vehicle accessibilities if the keyword does not match with any data 1`] = `
Object {
  "data": Array [],
}
`;

exports[`@positive get vehicle_accessibilities -- As a Store, I should be able to get all list of vehicle accessibilities 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "Mobil Box",
    },
    Object {
      "id": 2,
      "name": "Pick Up",
    },
    Object {
      "id": 3,
      "name": "Truck",
    },
    Object {
      "id": 4,
      "name": "Motor Bak",
    },
  ],
}
`;
